### PR TITLE
fix stuck screen

### PR DESCRIPTION
**Pull request by willmcgugan**
Fixes https://github.com/Textualize/textual/issues/1632

The fix was straightforward, but finding it required adding a few tools that are worth keeping.

A notable change is that if you mount anything without async, it is guaranteed to complete before the next message is processed. I think this will reduce race conditions where a dev doesn't await a mount.

- Fixes a stuck screen (deadlock on DOM lock)
- Adds a warning in debug mode when a callback takes too long to return
- Adds a `call_next` function which invokes a callback immediately after processing messages
- Forces awaiting of non-async mounts on the next message

### CODE REVIEW

File: src/textual/message_pump.py [hunk 1]
```diff
@@ -367,35 +381,52 @@ async def _process_messages_loop(self) -> None:
                 except MessagePumpClosed:
                     break
 
+            self._active_message = message
```
Comment (willmcgugan) [inline src/textual/message_pump.py:384]: Purely for debugging.  If a handler is stuck, we can see what message is being processed.

**Comment by davep** [comment]
Quickly testing the original issue here, 0.10.1 vs this PR and it looks good in terms if impact.

File: src/textual/widget.py [hunk 2]
```diff
@@ -91,6 +91,10 @@ def __init__(self, parent: Widget, widgets: Sequence[Widget]) -> None:
         self._parent = parent
         self._widgets = widgets
 
+    async def __call__(self) -> None:
```
Comment (rodrigogiraoserrao) [inline src/textual/widget.py:94]: Is this so we can use `invoke` to await this?

**Review by rodrigogiraoserrao — APPROVED** [review]
The changes make sense to me.
Async isn't my cup of tea yet, though.

**Review by davep — APPROVED** [review]

**Comment by willmcgugan** [comment]
> The changes make sense to me. Async isn't my cup of tea yet, though.

It's an acquired taste. Like blue cheese.

File: src/textual/widget.py (hunk 2 above)
Comment (willmcgugan) [reply src/textual/widget.py:94]: precisely.

**Pull request merged**